spec/unit/network/authconfig_spec.rb

```diff
@@ -107,3 +107,29 @@
     end
   end
 end
+
+describe Puppet::Network::AuthConfig do
+  class TestAuthProvider
+    def initialize(rights=nil); end
+    def check_authorization(method, path, params); end
+  end
+
+  it "instantiates authprovider_class with rights" do
+    Puppet::Network::AuthConfig.authprovider_class = TestAuthProvider
+    rights = Puppet::Network::Rights.new
+    TestAuthProvider.expects(:new).with(rights)
+    described_class.new(rights)
+  end
+
+  it "delegates authorization check to authprovider_class" do
+    Puppet::Network::AuthConfig.authprovider_class = TestAuthProvider
+    TestAuthProvider.any_instance.expects(:check_authorization).with(:save, '/path/to/resource', {})
+    described_class.new.check_authorization(:save, '/path/to/resource', {})
+  end
+
+  it "uses DefaultAuthProvider by default" do
+    Puppet::Network::AuthConfig.authprovider_class = nil
```

Comment:
Wasn't sure if you were implying this, but just in case, I wouldn't rely on the order of tests in the rspec group, e.g. if you run with rspec --order:random it can reorder tests, so "instantiates authprovider_class with rights" could come after "delegates authorization check to authprovider_class".